### ISSUE TITLE
Possible problem with treatment coding

### ISSUE THREAD

**Issue by vb690** [bambinos/bambi#334]
Hello :)

After updating bambi to the latest stable version, I noticed that my previous treatment coding with specified reference does not work anymore.  This 
```python
  traces = model.fit(
      "value ~ C(Model, Treatment('my_reference_level')) + 1|C(context)",
      draws=1000,
      tune=1000,
      priors=priors
  )
```
produce the following error
```
NameError: name 'Treatment' is not defined
```
I believe it's because my specification doesn't fit with what required by the `formulae` library. However I struggle to find on `formulae` GitHub page how to specify the reference level for treatment coding.

Any insight on this?

Btw thank you for the very nice library.

**Comment by tomicapretto** [comment]
Hi! Thanks for opening the issue. 

I'm sorry for the bad documentation. `formulae` is still very new and I couldn't add more documentation yet.

If I understand correctly, you're trying to use treatment encoding using 'my_reference_level' as reference level (instead of default reference level), right?
We are not supporting exactly the same as Treatment encoding in Patsy, but here's a workaround you can use. 

Suppose you have a variable called `var` with levels [0, 1, 2, 3, 4] and you want to use 3 as a reference level. You can do 

```python
lvls = [3, 0, 1, 2, 4] # or any other ordering. 
"y ~ C(var, levels=lvls)"
```
The first level you indicate in the `levels` argument is taken as a reference level.

Also, I see you are using mixed-effects. Just remember to use parenthesis for the random component so the parser understands that the LHS is `1` and not `C(Model, Treatment('my_reference_level')) + 1`. 

To sum up, this is what you could try

```python
lvls = ['my_reference_level', 'other_level_1', ....]
  traces = model.fit(
      "value ~ C(Model, levels=lvls) + (1|C(context))",
      draws=1000,
      tune=1000,
      priors=priors
  )
```
Let me know if it works and please leave any questions you may have about it. 
Hope it helps!

**Comment by vb690** [comment]
Hello :)

works like a charm! 
No apologies needed and thank you very much for your help.

I will close the issue.

**Comment by tomicapretto** [comment]
Another option is to convert the column in the data frame to an ordered categorical, i.e., 

```python
type_ = pd.api.types.CategoricalDtype(categories=['C', 'B', 'A'], ordered=True)
data['g'] = data['g'].astype(type_)
model = bmb.Model("y ~ g", data)
```
because `formulae` keeps the order given in the type (if `dtype.ordered == True`)
